src/katgpucbf/xbgpu/engine.py

```diff
@@ -206,17 +205,29 @@ async def sender_loop(self) -> None:
         """Send heaps to the network in a continuous loop.
 
         This function should do the following:
-        * Get a TxQueueItem from the tx_item_queue
-        * Ensure it is not a NoneType value (indicating shutdown sequence)
-        * Wait for events on the item to complete (likely GPU processing)
-        * Wait for an available heap buffer from the send_stream
-        * Asynchronously Transfer/Download GPU buffer data into the heap buffer in system RAM
-            * Wait for the transfer to complete, before
-        * Transmit heap buffer onto the network
-        * Place the TxQueueItem back on the tx_free_item_queue once complete
+        - Get a TxQueueItem from the tx_item_queue
+        - Ensure it is not a NoneType value (indicating shutdown sequence)
+        - Wait for events on the item to complete (likely GPU processing)
+        - Wait for an available heap buffer from the send_stream
+        - Asynchronously Transfer/Download GPU buffer data into the heap buffer in system RAM
+
+            - Wait for the transfer to complete, before
+
+        - Transmit heap buffer onto the network
+        - Place the TxQueueItem back on the tx_free_item_queue once complete
         """
         raise NotImplementedError  # pragma: nocover
 
+    @abstractmethod
+    def capture_start(self) -> None:
+        """Enable the transmission of this data product's stream."""
+        raise NotImplementedError  # pragma: nocover
+
+    @abstractmethod
+    def capture_stop(self) -> None:
+        """Disable the transmission of this data product's stream."""
+        raise NotImplementedError  # pragma: nocover
```

Comment:
You could probably reduce the amount of boilerplate you deal with by combining these into one function that takes a boolean.